schema/deploy/shipping/views.sql

```diff
@@ -2328,6 +2328,196 @@ grant select
   to "incidence-modeler";
 
 
+create or replace view shipping.scan_encounters_with_best_available_vaccination_data_v1 as
+
+    with vac_1_most_frequent_valid_date as (
+      select distinct on (individual)
+      individual,
+      vac_date,
+      vac_name_1
+      from shipping.scan_encounters_v1
+      where vac_date >= '2020-12-01'
+        and vac_date::date <= encountered::date
+      group by individual, vac_date, vac_name_1
+      having count(*) > 1
+      order by individual, count(*) desc
+    ),
+    vac_1_next_best_date as (
+      select distinct on (individual)
+      individual,
+      vac_date,
+      vac_name_1,
+      case
+        when vac_date < '2020-12-01' or vac_date::date > encountered::date then 1
+      end as vac_date_1_out_of_range
+      from shipping.scan_encounters_v1
+      where individual not in (select individual from vac_1_most_frequent_valid_date)
+        and vac_date is not null
+      order by individual, vac_date_1_out_of_range nulls first, encountered
+    ),
```

Comment:
To document conversation, we discussed whether these CTEs could be combined to simplify the query, but because of requirements to prefer the most frequently entered valid date, but to prefer less frequently invalid dates if thats all thats available, it makes sense to keep these separate.